<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NetworkUtilsActivity</class>
 <widget class="QWidget" name="NetworkUtilsActivity">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1478</width>
    <height>1044</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="tabSend">
      <attribute name="title">
       <string>Send</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="TextEntry" name="lineEditRemoteAddress">
           <property name="placeholderText">
            <string>test address</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="TextEntry" name="lineEditRemotePort">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maxLength">
            <number>6</number>
           </property>
           <property name="placeholderText">
            <string>test port</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Packet count:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="horizontalSliderSendCount">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>100</number>
           </property>
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="tickPosition">
            <enum>QSlider::TickPosition::TicksBelow</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QHexEdit" name="hexEditor">
         <property name="frameShape">
          <enum>QFrame::Shape::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Raised</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>size:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="horizontalSliderSendSize">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>100</number>
           </property>
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="tickPosition">
            <enum>QSlider::TickPosition::TicksBelow</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="pushButtonSendData">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="toolTip">
          <string>Send some test data to address and port selected</string>
         </property>
         <property name="text">
          <string>Send Data</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>838</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabConnection">
      <attribute name="title">
       <string>Connection</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBoxConnect">
         <property name="title">
          <string>Connect</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QPushButton" name="pushButtonTest">
              <property name="text">
               <string>TEST</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="LocalAddressEntry" name="comboBoxLocalAddress">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="TextEntry" name="lineEditBindPort">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Select on which port this hub should listen for incomming connections by remotes and agents</string>
              </property>
              <property name="maxLength">
               <number>6</number>
              </property>
              <property name="placeholderText">
               <string>Listen port</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="TryToggle" name="tryToggleListen" native="true">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>10</width>
                <height>10</height>
               </size>
              </property>
              <property name="toolTip">
               <string>Click to toggle server listening</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Orientation::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>897</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TextEntry</class>
   <extends>QLineEdit</extends>
   <header location="global">components/TextEntry.hpp</header>
  </customwidget>
  <customwidget>
   <class>QHexEdit</class>
   <extends>QFrame</extends>
   <header location="global">hexedit/QHexEdit.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TryToggle</class>
   <extends>QWidget</extends>
   <header location="global">components/TryToggle.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>LocalAddressEntry</class>
   <extends>QComboBox</extends>
   <header location="global">node/LocalAddressEntry.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <slots>
  <slot>confirmed()</slot>
  <slot>canceled()</slot>
 </slots>
</ui>
